test/test_sparse.py

```diff
@@ -1611,6 +1611,92 @@ def test_shape(i_shapes, v_shapes, nnzs):
         test_shape([0, 3, 4], [3, 4, 5, 6], [0])
         test_shape([2, 3, 4], [0, 4, 5, 6], [9, 12])
 
+        sparse_tensor, _, _ = self._gen_sparse(len([2, 3, 4]), 9, [2, 3, 4] + [3, 4, 5, 6])
+        for mem_format in [torch.channels_last, torch.channels_last_3d, torch.contiguous_format, torch.preserve_format]:
+            with self.assertRaisesRegex(RuntimeError, "memory format option is only supported by strided tensors"):
+                result = torch.zeros_like(sparse_tensor, memory_format=mem_format)
+
+        with self.assertRaisesRegex(
+            RuntimeError, r"Could not run 'aten::empty_strided' with arguments from the 'Sparse(CPU|CUDA)' backend"
+        ):
+            dense_tensor = sparse_tensor.to_dense()
+            result = torch.zeros_like(dense_tensor, layout=torch.sparse_coo)
+
+        result = torch.zeros_like(sparse_tensor, layout=torch.strided)
```

Comment:
Just pull these lines into the block where you enumerate the memory formats, here and with empty_like. 

```
sparse_tensor, _, _ = self._gen_sparse(len([2, 3, 4]), 9, [2, 3, 4] + [3, 4, 5, 6])
for mem_format in [torch.channels_last, torch.channels_last_3d, torch.contiguous_format, torch.preserve_format]:
  with self.assertRaisesRegex(RuntimeError, "memory format option is only supported by strided tensors"):
    result = torch.zeros_like(sparse_tensor, memory_format=mem_format)

  result = torch.zeros_like(sparse_tensor, layout=torch.strided, memory_format=mem_format)
  self.assertTrue(result.layout == torch.strided)
```